src/api_manager/check_auth.cc

```diff
@@ -160,10 +160,11 @@ void AuthChecker::Check() {
     Unauthenticated("Missing or invalid credentials");
     return;
   }
-  context_->request()->SetAuthToken(auth_token_);
 
-  env_->LogDebug(std::string("auth token: ") + auth_token_);
+  context_->request()->SetAuthToken(auth_token_);
 
+  //Set auth token for AuthzCache key construction in check_security_rules.cc.
+  context_->SetAuthToken(auth_token_);
```

Comment:
I added SetAuthToken function to /include/api_manager/request.h, /src/nginx/request.h, /src/nginx/request.cc, and /src/api_manager/mock_request.h. Now getting auth token would be a simple call to context_->request()->GetAuthToken().